### DISCUSSION TITLE
#3. VATZ biweekly meeting at 2022-03-06 13:00:00(KST) | 2022-03-15 10:00:00(CST)

### DISCUSSION THREAD

**Discussion by xellos00** [dsrvlabs/vatz#19] [Ideas]
## Remind previous biweekly meeting 

- Reminder
   - https://github.com/dsrvlabs/vatz/discussions/8#discussioncomment-2277819
   - https://github.com/dsrvlabs/vatz/discussions/8#discussioncomment-2278156

- TODOs & what's been decided 
   - https://github.com/dsrvlabs/vatz/discussions/8#discussioncomment-2278297

---

# Agenda for Meeting

- How initial Vatz works and its structure

- How to develop plugin per protocol

- How to manage Configuration

- How to deploy until CI/CD Pipeline is settled. 

---

**Comment by xellos00** [comment]
# Agenda

- Protobuffer
   - No minor Version (v1 -> v2)  버전의 바뀔 때는 대공사 (그러므로 하위 호환성은 무시한다)  
   - Dist 구조는 더 단순화 해서, 
      - 예제) github.com/dsrvlabs/vatz-proto/plugin/v1
      - 예제) github.com/dsrvlabs/vatz-proto/manager/v1

- Config
- Structures
- Test

**Reply by rootwarp** [reply]
- github.com/dsrvlabs/vatz-proto/plugin/v1
   - github.com/dsrvlabs/vatz-proto/manager(?)/v1

**Comment by xellos00** [comment]
<img width="1677" alt="image" src="https://user-images.githubusercontent.com/6308023/157804771-3e50bee4-780e-4de3-bf06-632bfbe952d1.png">

**Reply by xellos00** [reply]
Link: https://drive.google.com/file/d/1T5VEJElcboFbScXvlLRpePEJGK-f5g5O/view?usp=sharing

**Comment by rootwarp** [comment]
@xellos00 Hey.
I share simple diagram for next discussion.
Also I already shared diagram on Google drive. If you have no access, remind me.

https://drive.google.com/drive/folders/1fc61d_N_hJXHyXV7yCEsXhkXbYjPMJLl?usp=sharing

![Vatz drawio](https://user-images.githubusercontent.com/9974087/157656277-56953dc1-817f-498b-9933-455da06a9ddd.png)

**Comment by rootwarp** [comment]
@xellos00 After reviewing and fix the architecture, share some part with me.
Then we can test how we work together.

**Reply by xellos00** [reply]
I will post what's we have discussed yesterday.

**Comment by xellos00** [comment]
Meeting 2022-03-14
 
> As yesterday's discussion with @rootwarp 
## 1.Simplified architecture 

## 1.1 Vatz-proto

```
.
├── LICENSE
├── Makefile
├── README.md
├── manager
│   └── v1
│       ├── manager.pb.go
│       └── manager_grpc.pb.go
├── plugin
│   └── v1
│       ├── plugin.pb.go
│       └── plugin_grpc.pb.go
└── proto
    └── vatz
        ├── manager
        │   └── v1
        │       └── manager.proto
        └── plugin
            └── v1
                └── plugin.proto

```
- https://github.com/dsrvlabs/vatz-proto/pull/3

## 1.2. Vatz
```.
.
├── Makefile
├── README.md
├── default.yaml
├── go.mod
├── go.sum
├── main.go
└── manager
    ├── api
    │   ├── api.go
    │   └── api_test.go
    ├── config
    │   ├── config.go
    │   └── config_manager.go
    ├── executor
    │   ├── executor.go
    │   └── executor_manager.go
    ├── healthcheck
    │   ├── healthcheck.go
    │   └── healthcheck_manager.go
    └── notification
        ├── notification.go
        └── notification_manager.go


```

- https://github.com/dsrvlabs/vatz/pull/22

-------- 

Tasks for more ppl

Phase | Arch| TODOs | Comment |
:--|:--|:--|:--:|
Running-Pilot |  - Main -> Manager -> Component  <br> - API -> Main -> Manager -> Component |   - Library access private. <br> - Read Confi.yaml from github when server is running <br> - Set client to get connect as static  <br> - Let config Parser parsing the items <br> - Send a verifying <br> - Execute plugin API to call up  <br> |  There more ppl to join on this

Vatz
- API
  - api 
- Component
   - Config 
   - Notification
   - executor
   - health
- Manager
   - Config_Manager 
   - Notification_Manager
   - Executor_Manager
   - Health_Manager 
- Main

 ----

Plugin
@rootwarp, will create a sample SDK 
  - Create plugin SDK 

-------

Therefore We need to set new assignee for each of job under here 
| No | Tasks | Assignee | Due Date
|:--|:--|:--:|:--:|
1| Access git private repo for library  | @hqueue |  2022-03-29 (KST)
2| Read confi.yaml from github when override default.yaml  |  @xellos00 |  2022-03-29 (KST)
3| set Client to get connect to static so doesn't need to set a client again. | @xellos00 |  2022-03-29 (KST)
4| Config Parser | @xellos00 |  2022-03-29 (KST)
5| Send a verifying |  @meetrick |  2022-03-29 (KST)
6| Execute plugin API to Plugin |  @heejin-github  |  2022-03-29 (KST)
7| Send a Notification to Discord |  @Choi-Jinhong  |  2022-03-29 (KST)
8| Plugin SDK | @rootwarp |  2022-03-29 (KST)
